thirdparty/runtime_environments.go

```diff
@@ -347,6 +365,89 @@ func stringToTime(timeInitial string) (time.Time, error) {
 	return time.Unix(timestamp, 0), nil
 }
 
+// Image stores information about an image including its name, version_id, kernel, ami, and its last deployed time.
+type Image struct {
+	Name         string
+	VersionID    string
+	Kernel       string
+	LastDeployed time.Time
+	AMI          string
+}
+
+// getNameFromOSInfo uses the provided AMI and name (exact match) arguments to filter the image information.
+func (c *RuntimeEnvironmentsClient) getNameFromOSInfo(ctx context.Context, ami string, name string) (string, error) {
+	opts := OSInfoFilterOptions{
+		AMI:  ami,
+		Name: fmt.Sprintf("^%s$", name),
+	}
+	result, err := c.GetOSInfo(ctx, opts)
+	if err != nil {
+		return "", errors.Wrap(err, "getting OS info")
+	}
+	if len(result) == 0 {
+		return "", errors.Errorf("OS information name '%s' not found for distro", opts.Name)
+	} else if len(result) > 1 {
+		return "", errors.Errorf("multiple results found for OS information name '%s'", opts.Name)
+	}
+	return result[0].Version, nil
+}
+
+// getLatestImageHistory returns the latest AMI and timestamp given the provided imageId.
+func (c *RuntimeEnvironmentsClient) getLatestImageHistory(ctx context.Context, imageID string) (*ImageHistoryInfo, error) {
+	optsHistory := ImageHistoryFilterOptions{
+		ImageID: imageID,
+		Limit:   1,
+	}
+	resultHistory, err := c.getHistory(ctx, optsHistory)
+	if err != nil {
+		return nil, errors.Wrapf(err, "getting history for image '%s': '%s'", imageID, err.Error())
+	}
+	if len(resultHistory) == 0 {
+		return nil, errors.Errorf("history for image '%s' not found", imageID)
+	} else if len(resultHistory) > 1 {
+		return nil, errors.Errorf("found more history than history limit provided for image '%s'", imageID)
```

Comment:
I think I asked for this and it seems fine because it's just in case something in the db goes wrong or something 

but I would say in the error 'expected 1 history but got %d' or something like that

Reply:
Agree - deleted